src/schedule/zephyr_dma_domain.c

```diff
@@ -0,0 +1,298 @@
+// SPDX-License-Identifier: BSD-3-Clause
```

Comment:
I would call this zephyr_preempt_domain.c because the code should be reusable for timers too to schedule processing that is not low latency. i.e. media decode with large buffers.

Reply:
@lgirdwood sorry, could you explain? As far as I understand the SOF LL scheduler has a "domain" abstraction which is a mechanism that triggers the LL scheduler scheduling function upon an event. One of those domains is a timer domain. This PR introduces a new *alternative* domain - a DMA domain. How can "timers" also use it? Isn't it decided with a task granularity - you initialise a task with a specific domain and then that domain handles the task?

Reply:
@lyakh I mean the timer should also be able to schedule work in the preemptive domain (alongside the LL domain).